### PR TITLE
Update trial status DB operation

### PR DESCRIPTION
**Pull request by hougangliu**
Part of #510

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/537)

### CODE REVIEW

File: pkg/controller/v1alpha2/experiment/experiment_controller.go [hunk 1]
```diff
@@ -233,7 +232,6 @@ func (r *ReconcileExperiment) Reconcile(request reconcile.Request) (reconcile.Re
 			logger.Error(err, "Create experiment in DB error")
 			return reconcile.Result{}, err
 		}
-		requeue = true
```
Comment (johnugeorge) [inline pkg/controller/v1alpha2/experiment/experiment_controller.go]: why is this removed?

File: pkg/controller/v1alpha2/experiment/experiment_controller.go [hunk 2]
```diff
@@ -257,7 +255,7 @@ func (r *ReconcileExperiment) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
-	return reconcile.Result{Requeue: requeue}, nil
```
Comment (johnugeorge) [inline pkg/controller/v1alpha2/experiment/experiment_controller.go]: Same comment?

File: pkg/controller/v1alpha2/experiment/experiment_controller.go (hunk 1 above)
Comment (hougangliu) [reply pkg/controller/v1alpha2/experiment/experiment_controller.go]: in this loop, instance status changed or fails(line 234). for status changed case, there is no need to requeue since changes of obj will make it into work queue again; for db write fail case, the obj key will not be pop out of the queue, similar as `Requeue: true`

File: pkg/controller/v1alpha2/experiment/experiment_controller.go (hunk 2 above)
Comment (hougangliu) [reply pkg/controller/v1alpha2/experiment/experiment_controller.go]: after my change, `requeue` will always false, and reconcile.Result.Requeue is 'false' by default

File: pkg/controller/v1alpha2/experiment/experiment_controller.go (hunk 1 above)
Comment (johnugeorge) [reply pkg/controller/v1alpha2/experiment/experiment_controller.go]: Can you check if reconcile function will be called immediately or wait for the sync period?

Comment (hougangliu) [reply pkg/controller/v1alpha2/experiment/experiment_controller.go]: if `Requeue` is true even update done, it will be called immediately. Then we calls Get operation, the return nearly always is obsolete data (That's why I find I need do this change).

**Comment by gaocegege** [comment]
Please rebase the master, Thanks
